core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-reports/store/dot-experiments-reports-store.ts

```diff
@@ -179,23 +193,41 @@ export class DotExperimentsReportsStore extends ComponentStore<DotExperimentsRep
             : null
     );
 
-    readonly getDetailData$: Observable<DotExperimentDetail[]> = this.select(({ results }) =>
-        results
-            ? Object.values(results.goals.primary.variants).map((variant) => ({
-                  id: variant.variantName,
-                  name: variant.variantDescription,
-                  trafficSplit: 'TBD',
-                  pageViews: variant.totalPageViews,
-                  sessions: results.sessions.variants[variant.variantName],
-                  clicks: variant.uniqueBySession.count,
-                  bestVariant: variant.uniqueBySession.totalPercentage / 100,
-                  improvement:
-                      (variant.uniqueBySession.totalPercentage -
-                          results.goals.primary.variants.DEFAULT.uniqueBySession.totalPercentage) /
-                      100,
-                  isWinner: results.bayesianResult.suggestedWinner === variant.variantName
-              }))
-            : []
+    readonly getDetailData$: Observable<DotExperimentDetail[]> = this.select(
+        ({ experiment, results }) => {
+            const hasBayesianResults = results?.bayesianResult?.results?.length > 0;
+            const noData = this.dotMessageService.get(NOT_ENOUGH_DATA_LABEL);
+
+            return results
+                ? Object.values(results.goals.primary.variants).map((variant) => {
+                      const variantBayesianResult = this.getBayesianVariantResult(
+                          variant.variantName,
+                          results.bayesianResult.results
+                      );
+
+                      return {
+                          id: variant.variantName,
+                          name: variant.variantDescription,
+                          conversions: variant.uniqueBySession.count,
+                          conversionRate: this.getConversionRate(
+                              variant.uniqueBySession.count,
+                              results.sessions.variants[variant.variantName]
+                          ),
+                          conversionRateRange: hasBayesianResults
+                              ? this.getConversionRateRage(
+                                    variantBayesianResult.credibilityInterval
+                                )
+                              : noData,
+                          sessions: results.sessions.variants[variant.variantName],
+                          probabilityToBeBest: hasBayesianResults
+                              ? formatPercent(variantBayesianResult.probability, 'en-US', '1.0-2')
+                              : noData,
+                          isWinner: results.bayesianResult.suggestedWinner === variant.variantName,
+                          isPromoted: this.isPromotedVariant(experiment, variant.variantName)
+                      };
```

Comment:
Looks like a complex object, can we make a function for this? Maybe we can even split it in even more functions to reduce the cognitive complexity like:

`getConversionRateRange() getProbabilityToBeBest()`

Reply:
done ! @zJaaal